src/runtime/entitlements-manager.js

```diff
@@ -116,6 +116,15 @@ export class EntitlementsManager {
       this.responsePromise_ = this.getEntitlementsFlow_(encryptedDocumentKey);
     }
     return this.responsePromise_.then(response => {
+      if (
```

Comment:
Do you want to do this here or in AMP?

Reply:
I think here would be easier since we make multiple getEntitlements calls in AMP, whereas all getEntitlements calls go through this method in SwG.

Reply:
But we still want to do something very similar in AMP as well for all other platforms, right? Otherwise, from the pov of SwG we here essentially testing ourselves. I'm pretty confident once we launch if we ever get a granting entitlement w/o decryption key it'd be purely due to a bug on our side and not a user error.

Reply:
Truth. I will update AMP too. Thanks for the heads up.